### DISCUSSION TITLE
Is using .global component styles officially supported?

### DISCUSSION THREAD

**Discussion by jtlindsey** [nuxt/nuxt#23985] [Questions]
I'm using the .global component style (naming some files as `[CompName].global.vue` for automatic global registration) for some dynamically registered components. The only documentation i could find about this exact syntax is [here](https://github.com/nuxt/nuxt/issues/14623#issuecomment-1397361562) and couldn't find it in the official [docs](https://nuxt.com/docs/guide/directory-structure/components#dynamic-components)

Is it officially supported but not documented, or is this an ancillary benefit that may go away without notice?

**Comment by TheAlexLichter** [comment]
IMO, nowadays the `~/components/global` folder is preferred

**Reply by jtlindsey** [reply]
How do you keep up with where things are in large apps when you have some components organized differently? In my case the global file is only rendered in respect to a mega component and is the only part of that component that is dynamically rendered.

Here is what both options might look like in a components tree:

```
.
├── components/
│   ├── MegaCompA/
│   │   ├── Bar.vue
│   │   ├── Buzz.vue
│   │   ├── Fiz.vue
│   │   ├── Foo.vue
│   │   ├── ShakeA.vue
│   │   └── ShakeB.global.vue <---inline method...
│   ├── MegaCompB/
│   │   ├── A.vue
│   │   ├── B.vue
│   │   ├── C.vue
│   │   └── D.vue
│   ├── Etc
│   ├── ...
│   ├── ...
│   ├── ...
│   └── global/
│       ├── ...
│       ├── ...
│       ├── ShakeA.vue <---or file from MegaCompA?
│       └── ...
└── *
```
